test/jasmine/tests/frame_api_test.js

```diff
@@ -0,0 +1,205 @@
+var Plotly = require('@lib/index');
+
+var createGraphDiv = require('../assets/create_graph_div');
+var destroyGraphDiv = require('../assets/destroy_graph_div');
+
+// A helper function so that failed tests don't simply stall:
+function fail(done) {
```

Comment:
Ah, `assets/fail_test.js` is present. Syntax is `.then(...).catch(fail).then(done)`, which is perfectly fine, maybe a bit cleaner and simpler.